SU2_CFD/src/solvers/CTurbSolver.cpp

```diff
@@ -674,21 +673,15 @@ void CTurbSolver::ComputeUnderRelaxationFactor(CSolver **solver_container, const
         /* We impose a limit on the maximum percentage that the
          turbulence variables can change over a nonlinear iteration. */
 
-        const unsigned long index = iPoint*nVar + iVar;
-        su2double ratio = LinSysSol[index]/(nodes->GetSolution(iPoint, iVar)+EPS);
-        if (ratio > allowableIncrease) {
-          localUnderRelaxation = min(allowableIncrease/ratio, localUnderRelaxation);
-        } else if (ratio < allowableDecrease) {
-          localUnderRelaxation = min(fabs(allowableDecrease)/ratio, localUnderRelaxation);
+        const unsigned long index = iPoint * nVar + iVar;
+        su2double ratio = fabs(LinSysSol[index]) / (fabs(nodes->GetSolution(iPoint, iVar)) + EPS);
+        if (ratio > allowableRatio) {
+          localUnderRelaxation = min(allowableRatio / ratio, localUnderRelaxation);
         }
 
       }
     }
 
-    /* Choose the minimum factor between mean flow and turbulence. */
-
-    localUnderRelaxation = min(localUnderRelaxation, solver_container[FLOW_SOL]->GetNodes()->GetUnderRelaxation(iPoint));
```

Comment:
I guess some numerical issues might occur if numbers get too large/small during convergence, although that sounds more like a single precision consideration.

Is it worth keeping just this min() between flow and turbulence for now? And maybe investigate if it makes sense as part of #1036?

Reply:
Fair enough. I'll add it back in for now since it shouldn't hurt anything